src/grid/grid.component.ts

```diff
@@ -0,0 +1,630 @@
+    protected selectedRow: any;
+    private filteringExpressions: Object = {};
+    private sortingExpressions: Object = {};
+    private colDiffer: Subscription;
+    private _differ: IterableDiffer;
+    private _paging: boolean = false;
+    private _lastRow: any;
+    private shouldDataBind: boolean = false;
+
+    constructor(private _renderer: Renderer,
+                private _elementRef: ElementRef,
+                private _detector: ChangeDetectorRef,
+                private _itDiff: IterableDiffers,
+                private _resolver: ComponentFactoryResolver,
+                private _viewRef: ViewContainerRef) {
+                    this._differ = this._itDiff.find([]).create(null);
+                }
+
+    public ngOnInit(): void {
+        if (this.data) {
+            this.dataContainer = new DataContainer(this.data);
+            if (this.paging) {
+                this.dataContainer.state.paging = {
+                    index: 0,
+                    recordsPerPage: this.rowsPerPage,
+                };
+            }
+            // should we attach to all browsers and only when we have sorting??
+            if (this.hasSorting) {
+                this.dataContainer.state.sorting.strategy = new StableSortingStrategy();
+            }
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public ngAfterContentInit(): void {
+        if (this.data && this.autoGenerate) {
+            this.autogenerate();
+        } else {
+            this.checkColumns();
+        }
+        this.colDiffer = this.cols.changes.subscribe((_) => {
+            this.checkColumns();
+            this._detector.markForCheck();
+        });
+
+    }
+
+    /**
+     * @hidden
+     */
+    protected checkColumns(): void {
+        this.columns = this.cols.toArray();
+        this.columns.forEach((col, index) => {
+            col.index = index;
+            this.onColumnInit.emit({column: col});
+        });
+    }
+
+    /**
+     * @hidden
+     */
+    public ngDoCheck(): void {
+        let diff: IterableDiffer = this._differ.diff(this.data);
+        if (this.shouldDataBind || diff) {
+            this.onBeforeProcess.emit(this.dataContainer.state);
+            this.dataContainer.process();
+            this.shouldDataBind = false;
+        }
+    }
+
+    /**
+     * @hidden
+     */
+    public ngOnDestroy(): void {
+        this.colDiffer.unsubscribe();
+    }
+
+    public getColumnByIndex(index: number): any {
+        if (this.columns[index] !== undefined) {
+            return this.columns[index];
+        }
+    }
+
+    public getColumnByField(field: string): any {
+        return this.columns.find((col) => col.field === field);
+    }
+
+    /**
+     * Returns the cell at rowIndex/columnIndex.
+     *
+     * @param {number} rowIndex
+     * @param {number} columnIndex
+     * @returns
+     *
+     * @memberOf IgxGrid
+     */
+    public getCell(rowIndex: number, columnIndex: number): any {
```

Comment:
Also there should be a `GridCell` type, same as with `getRow`, and the cell should consist of element and dataItem.